IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetry.m

```diff
@@ -135,4 +197,51 @@ - (MSIDLastRequestTelemetrySerializedItem *)createSerializedItem
     return [[MSIDLastRequestTelemetrySerializedItem alloc] initWithSchemaVersion:[NSNumber numberWithInteger:self.schemaVersion] defaultFields:defaultFields errorInfo:self.errorsInfo platformFields:nil];
 }
 
+#pragma mark - Private: Save To Disk
+
+- (void)saveTelemetryToDisk
+{
+    dispatch_barrier_sync(self.synchronizationQueue, ^{
```

Comment:
just curious why do we need to synchronize here since file writing should be sync already, no?